CMakeLists.txt

```diff
@@ -9,21 +9,24 @@ project(yaLanTingLibs
 # load pack finder
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/Find/)
 
-# add include path
-include_directories(include)
-include_directories(include/ylt/thirdparty)
-include_directories(src/include)
-
 find_package(Threads REQUIRED)
 link_libraries(Threads::Threads)
 
-include(cmake/utils.cmake)
-include(cmake/struct_pb.cmake)
-include(cmake/build.cmake)
-include(cmake/develop.cmake)
 include(cmake/install.cmake)
-# add project config, such as enable_ssl.
-include(cmake/config.cmake)
-# add project's source such as unit test, example & benchmark
-include(cmake/subdir.cmake)
+
+if(PROJECT_IS_TOP_LEVEL) # ref:https://cmake.org/cmake/help/latest/variable/PROJECT_IS_TOP_LEVEL.html
```

Comment:
this trick may helpful if we do not use `PROJECT_IS_TOP_LEVEL`

https://github.com/google/googletest/blob/b3a9ba2b8e975550799838332803d468797ae2e1/googletest/CMakeLists.txt#L57C10-L57C10